File: 3734-346-643-maximum-average-subarray-i/3734-346-643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int maxSum = 0,interSum = 0;
        for(int i=0;i<k;i++)
            maxSum+= nums[i];
        interSum = maxSum;
        for(int i=k;i<nums.length;i++){
            interSum+= nums[i]-nums[i-k];
            maxSum = Math.max(maxSum,interSum); 
        }
        return (double) maxSum/k;
    }
}

File: 3734-346-643-maximum-average-subarray-i/3734-346-643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        maxSum = 0
        for i in range(k):
            maxSum+= nums[i]
        
        interSum = maxSum

        for i in range(k,len(nums)):
            interSum+= nums[i] - nums[i-k]
            maxSum = max(interSum,maxSum)

        return maxSum / k